gulpfile.js

```diff
@@ -36,15 +36,295 @@ var fs = require('fs');
 var rimraf = require('rimraf');
 var execSync = require('child_process').execSync;
 
-// Rebuilds Blockly, including the following:
-//  - blockly_compressed.js
-//  - blocks_compressed.js
-//  - Localization string tables in msg/js/*.js
-//  - Generators in generators/*.js
-// These files are already up-to-date in the master branch.
-gulp.task('build', gulp.shell.task([
-  'python build.py'
-]));
+var closureCompiler = require('google-closure-compiler').gulp();
+var packageJson = require('./package.json');
+var argv = require('yargs').argv;
+
+
+////////////////////////////////////////////////////////////
+//                        Build                           //
+////////////////////////////////////////////////////////////
+
+const licenseRegex = `\\/\\*\\*
+ \\* @license
+ \\* [\\w: ]+
+ \\*
+ \\* (Copyright \\d+ (Google Inc.|Massachusetts Institute of Technology))
+ \\* (https://developers.google.com/blockly/|All rights reserved.)
+ \\*
+ \\* Licensed under the Apache License, Version 2.0 \\(the "License"\\);
+ \\* you may not use this file except in compliance with the License.
+ \\* You may obtain a copy of the License at
+ \\*
+ \\*   http://www.apache.org/licenses/LICENSE-2.0
+ \\*
+ \\* Unless required by applicable law or agreed to in writing, software
+ \\* distributed under the License is distributed on an "AS IS" BASIS,
+ \\* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ \\* See the License for the specific language governing permissions and
+ \\* limitations under the License.
+ \\*\\/`;
+
+/**
+ * Helper method for stripping the Google's and MIT's Apache Licenses.
+ */
+function stripApacheLicense() {
+  // Strip out Google's and MIT's Apache licences.
+  // Closure Compiler preserves dozens of Apache licences in the Blockly code.
+  // Remove these if they belong to Google or MIT.
+  // MIT's permission to do this is logged in Blockly issue #2412.
+  return gulp.replace(new RegExp(licenseRegex, "g"), '');
+}
+
+/**
+ * Helper method for prepending the auto-generated header text.
+ */
+function prependHeader() {
+  return gulp.insert.prepend(`// Do not edit this file; automatically generated by build.py.\n`);
```

Comment:
Is the build.py part of this comment still valid?

Reply:
Yes, I wanted to keep the two compatible for now, but as I think about it more, maybe we should know the source of truth.